src/xenia/base/memory.cc

```diff
@@ -215,7 +219,102 @@ void copy_and_swap_16_in_32_unaligned(void* dest_ptr, const void* src_ptr,
     dest[i] = (src[i] >> 16) | (src[i] << 16);
   }
 }
+
+#elif XE_ARCH_ARM64
+
+// Although NEON offers vector rev instructions (like vrev32q_u8), they are
+// slower in benchmarks. Also, using uint8x16xN_t wasn't any faster in the
+// benchmarks, hence we use just use one SIMD register to minimize residual
+// processing.
+
+void copy_and_swap_16_aligned(void* dst_ptr, const void* src_ptr,
+                              size_t count) {
+  copy_and_swap_16_unaligned(dst_ptr, src_ptr, count);
+}
+
+void copy_and_swap_16_unaligned(void* dst_ptr, const void* src_ptr,
+                                size_t count) {
+  auto dst = reinterpret_cast<uint16_t*>(dst_ptr);
```

Comment:
`static_cast` is enough for `void*` <> pointer.

Reply:
`static_cast` can adjust the address when casting from `void*` to a pointer of any type, for example to satisfy alignment requirements. [see 10)](https://en.cppreference.com/w/cpp/language/static_cast)

`reinterpret_cast` is guaranteed (weird architectures aside) to not interfere with the pointer (not generating any cpu instructions)

even if it might be ok on aarch64 I vote to use `reinterpret_cast`. Because it's _"more correct"_ and code like this get's copied around and might end up in our planned Alpha AXP support and not be fine anymore :/